packages/surveycto/src/Adaptor.js

```diff
@@ -98,70 +100,239 @@ export function fetchSubmissions(formId, options = {}, callback = s => s) {
 
     console.log(`Fetching '${resolvedFormId}' submissions for: ${date}`);
 
-    return requestHelper(
-      state,
-      path,
-      {
-        headers: {
-          'content-type': contentType,
-        },
-        query: {
-          date,
-          r: status,
-        },
+    const response = await requestHelper(state, path, {
+      headers: {
+        'content-type': contentType,
       },
-      callback
+      query: {
+        date,
+        r: status,
+      },
+    });
+
+    return prepareNextState(state, response);
+  };
+}
+
+/**
+ * List resources from SurveyCTO
+ * @public
+ * @example <caption>List all dataset records</caption>
+ * list('datasets/enumerators_dataset/records')
+ * @example <caption>List all datasets</caption>
+ * list('datasets')
+ * @example <caption>List dataset records with pagination options</caption>
```

Comment:
`enumerators_dataset` is a dataset id, right?

I think these examples would be clearer like this (it would certainly help me!)
```
list(`datasets/${$.datasetId}/records`)
```